plans/query_plans.go

```diff
@@ -708,11 +727,32 @@ func (plan *QueryPlan) Select() ([]interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
+	if plan.cacheable {
+		cacheKey := fmt.Sprintf("%s: %v", query, plan.args)
+		data, err := getCacheData(cacheKey, plan.memCache)
+		if err == nil { // fail silently - graceful fallback
+			return data, nil
+		}
+	}
+
 	target := plan.target.Interface()
 	if subQuery, ok := target.(subQuery); ok {
 		target = subQuery.getTarget().Interface()
 	}
-	return plan.executor.Select(target, query, plan.args...)
+	res, err := plan.executor.Select(target, query, plan.args...)
+
```

Comment:
We need to check `err` here - if it's non-nil, we don't want to store the cache entry.